Rubberduck.VBEEditor/Class1.cs

```diff
@@ -0,0 +1,964 @@
+        {
+            ppv = IntPtr.Zero;
+            if ((!_isDisposed) && (iid == typeof(T).GUID))       // no need to offer IID_IUnknown here, as it is handled by the aggregation object
+            {
+                ppv = _outerObject;
+                Marshal.AddRef(_outerObject);
+                return CustomQueryInterfaceResult.Handled;
+            }
+            return CustomQueryInterfaceResult.Failed;
+        }
+    }
+
+    [ComImport(), Guid("00020400-0000-0000-C000-000000000046")]
+    [InterfaceType(ComInterfaceType.InterfaceIsIDispatch)]
+    public interface IDispatch
+    {
+    }
+
+    // A compatible version of ITypeInfo, where COM objects are outputted as IntPtrs instead of objects
+    [ComImport(), Guid("00020401-0000-0000-C000-000000000046")]
+    [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+    public interface ITypeInfo_VBE
+    {
+        void GetTypeAttr(out IntPtr ppTypeAttr);
+        void GetTypeComp(out IntPtr ppTComp);
+        void GetFuncDesc(int index, out IntPtr ppFuncDesc);
+        void GetVarDesc(int index, out IntPtr ppVarDesc);
+        void GetNames(int memid, [Out] out string rgBstrNames, int cMaxNames, out int pcNames);
+        void GetRefTypeOfImplType(int index, out int href);
+        void GetImplTypeFlags(int index, out IMPLTYPEFLAGS pImplTypeFlags);
+        void GetIDsOfNames(string[] rgszNames, int cNames, int[] pMemId);
+        void Invoke(object pvInstance, int memid, short wFlags, ref DISPPARAMS pDispParams, IntPtr pVarResult, IntPtr pExcepInfo, out int puArgErr);
+        void GetDocumentation(int index, out string strName, out string strDocString, out int dwHelpContext, out string strHelpFile);
+        void GetDllEntry(int memid, INVOKEKIND invKind, IntPtr pBstrDllName, IntPtr pBstrName, IntPtr pwOrdinal);
+        void GetRefTypeInfo(int hRef, out IntPtr ppTI);
+        void AddressOfMember(int memid, INVOKEKIND invKind, out IntPtr ppv);
+        void CreateInstance(object pUnkOuter, ref Guid riid, out object ppvObj);
+        void GetMops(int memid, out string pBstrMops);
+        void GetContainingTypeLib(out IntPtr ppTLB, out int pIndex);
+        void ReleaseTypeAttr(IntPtr pTypeAttr);
+        void ReleaseFuncDesc(IntPtr pFuncDesc);
+        void ReleaseVarDesc(IntPtr pVarDesc);
+
+        void Placeholder1();
+        IDispatch GetStdModInstance();            // a handy extra vtable entry we can use to invoke members in standard modules.
+    }
+
+    // FIXME there's probably some better builtin c# class for this
+    public class DisposableList<T> : List<T>, IDisposable
+        where T : class
+    {
+        public void Dispose() => ((IDisposable)this).Dispose();
+
+        private bool _isDisposed;
+        void IDisposable.Dispose()
+        {
+            if (_isDisposed) return;
+            _isDisposed = true;
+
+            foreach (var element in this)
+            {
+                ((IDisposable)element).Dispose();
+            }
+        }
```

Comment:
actually what one could try to do is override the List API to return default values when this instance has been Disposed...